spyder/plugins/ipythonconsole.py

```diff
@@ -1464,11 +1471,13 @@ def _create_client_for_kernel(self, connection_file, hostname, sshkey,
                                    _("Could not open ssh tunnel. The "
                                      "error was:\n\n") + to_text_string(e))
                 return
-        kernel_client.start_channels()
 
         # Assign kernel manager and client to shellwidget
         client.shellwidget.kernel_client = kernel_client
         client.shellwidget.kernel_manager = kernel_manager
+        client.sig_spyder_kernel.connect(self.connect_shellwidget)
+        kernel_client.start_channels()
+        client.shellwidget.is_spyder_kernel()

```

Comment:
The validation is precisely done here. We iterate over all open clients in the IPython console to see if we're trying to connect to the kernel of a current client. If that's the case, we don't need to connect that kernel to the Variable Explorer because it's already connected.

If that fails, then it means we're trying to connect to an external kernel, in which case we want to connect it to the Variable Explorer (if it's a Spyder kernel, of course).

What you've done was the right thing to do :-)